### PR TITLE
(BOLT-381) Bump minimum Ruby version to 2.3

### PR DESCRIPTION
**Pull request by MikaelSmith**
That allows us to use several improved language constructs.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Gem-based acceptance tests will probably need updated Ruby versions. I’ll look into it tomorrow.

File: lib/bolt/inventory.rb [hunk 1]
```diff
@@ -41,8 +43,7 @@ def self.default_paths
     def self.from_config(config)
       if ENV.include?(ENVIRONMENT_VAR)
         begin
-          # rubocop:disable YAMLLoad
-          data = YAML.load(ENV[ENVIRONMENT_VAR])
+          data = YAML.safe_load(ENV[ENVIRONMENT_VAR])
         # In older releases of psych SyntaxError is not a subclass of Exception
```
Comment (adreyer) [inline lib/bolt/inventory.rb]: We can remove Syntax error if we're calling safe_load

File: lib/bolt/util.rb [hunk 2]
```diff
@@ -15,10 +17,7 @@ def read_config_file(path, default_paths = nil, file_name = 'file')
         end
 
         path = File.expand_path(path)
-        # safe_load doesn't work with psych in ruby 2.0
-        # The user controls the configfile so this isn't a problem
-        # rubocop:disable YAMLLoad
-        File.open(path, "r:UTF-8") { |f| YAML.load(f.read) }
+        File.open(path, "r:UTF-8") { |f| YAML.safe_load(f.read) }
```
Comment (adreyer) [inline lib/bolt/util.rb]: We can remove rescue for Syntax error if we're calling safe_load

**Comment by MikaelSmith** [comment]
I think this is ready again. Had to fix up acceptance setup.

File: spec/bolt/transport/ssh_spec.rb [hunk 3]
```diff
@@ -106,7 +108,7 @@ def result_value(stdout = nil, stderr = nil, exit_code = 0)
       expect_node_error(Bolt::Node::ConnectError,
                         'CONNECT_ERROR',
                         /Failed to connect to/) do
-        ssh.with_connection(make_target(h: hostname, p: 65535)) {}
+        ssh.with_connection(make_target(port_: 65535)) {}
```
Comment (lucywyman) [inline spec/bolt/transport/ssh_spec.rb]: Does this have a default host_ now?

Comment (MikaelSmith) [reply spec/bolt/transport/ssh_spec.rb]: It always did. I can add that back, but it was unnecessary.

Comment (lucywyman) [reply spec/bolt/transport/ssh_spec.rb]: No need, was just curious

**Comment by MikaelSmith** [comment]
@lucywyman fixed and added a test.

File: bolt-modules/boltlib/lib/puppet/functions/run_plan.rb [hunk 4]
```diff
@@ -47,8 +49,7 @@ def run_plan(scope, plan_name, named_args = {})
       begin
         result = func.class.dispatcher.dispatchers[0].call_by_name_with_scope(scope, params, true)
       rescue Puppet::PreformattedError => err
-        if named_args['_catch_errors'] &&
-           err.respond_to?(:cause) && err.cause && err.cause.is_a?(Bolt::Error)
+        if named_args['_catch_errors'] && err.cause && err.cause.is_a?(Bolt::Error)
```
[the remark this answers is not in the document]
Comment (MikaelSmith) [reply bolt-modules/boltlib/lib/puppet/functions/run_plan.rb]: Ok, I'll do that.

**Comment by MikaelSmith** [comment]
Finally green.

**Pull request merged**